File: Problems/ICPC Asyut Sheets/Sheet #3 (Arrays)/L. Max Subarray.cpp
#include <bits/stdc++.h>
using namespace std;

int maxElement(int *arr, int start, int end)
{
    int max = arr[start];
    for (int i = start; i <= end; i++)
    {
        if (arr[i] > max)
            max = arr[i];
    }
    return max;
}
int main()
{
    int t, n;
    cin >> t;
    while (t--)
    {
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++)
            cin >> arr[i];

        for (int i = 0; i < n; i++)
        {
            for (int j = i; j < n; j++)
                cout << maxElement(arr, i, j) << " ";
        }
        cout << "\n";
    }
    return 0;
}

File: Problems/ICPC Asyut Sheets/Sheet #3 (Arrays)/L. Max Subarray.py
def max_element(arr, start, end):
    max_val = arr[start]
    for i in range(start, end + 1):
        if arr[i] > max_val:
            max_val = arr[i]
    return max_val

t = int(input())
for _ in range(t):
    n = int(input())
    arr = list(map(int, input().split()))

    for i in range(n):
        for j in range(i, n):
            print(max_element(arr, i, j), end=" ")
    print()